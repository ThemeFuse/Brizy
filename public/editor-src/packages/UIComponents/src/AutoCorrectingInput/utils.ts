import { clamp } from "../utils/math";

export function formatInputValue(value: number, step: number): string | number {
  const stepDecimalLength = decimalLength(step);

  return stepDecimalLength > 0
    ? value.toFixed(stepDecimalLength)
    : String(value);
}

export function decimalLength(number: number): number {
  const [, decimal] = String(number).split(".");
  return decimal ? decimal.length : 0;
}

export function correctNumber(
  number: number,
  min: number,
  max: number,
  step: number
): number {
  return roundByStep(clamp(number, min, max), step, min);
}

export const preciseAdd = preciseOperation.bind(null, "add");

export const preciseSub = preciseOperation.bind(null, "sub");

function roundByStep(number: number, step: number, offset = 0) {
  // this is done to avoid quirks with floating point operations
  // like 1.1 - 1 = 0.10000000000000009
  const mul = 10 ** decimalLength(step);

  number = number * mul;
  step = step * mul;
  offset = offset * mul;

  const tmp = Math.ceil((number - offset) / step) * step + offset;

  return tmp / mul;
}

function preciseOperation(type: "add" | "sub", a: number, b: number) {
  const dla = decimalLength(a);
  const dlb = decimalLength(b);
  const dlMax = Math.max(dla, dlb);
  const mul = 10 ** dlMax;

  const a_ = a * mul;
  const b_ = b * mul;
  let res;

  switch (type) {
    case "add":
      res = a_ + b_;
      break;
    case "sub":
      res = a_ - b_;
      break;
    default:
      throw new Error(`operation type unknown ${type}`);
  }

  return res / mul;
}
