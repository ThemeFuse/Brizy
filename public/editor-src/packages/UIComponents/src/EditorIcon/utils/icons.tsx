import {
  CheckCircle,
  CheckSmall,
  Circle2,
  CircleRemove,
  Close,
  Duplicate,
  Heart,
  MediaImage,
  MediaMap,
  MediaVideo,
  MinistryBrandsEventCalendar,
  MinistryBrandsEventDetail,
  MinistryBrandsEventFeatured,
  MinistryBrandsEventLayout,
  MinistryBrandsEventList,
  MinistryBrandsGroupDetail,
  MinistryBrandsGroupFeatured,
  MinistryBrandsGroupLayout,
  MinistryBrandsGroupList,
  MinistryBrandsGroupSlider,
  MinistryBrandsSermonDetail,
  MinistryBrandsSermonFeatured,
  MinistryBrandsSermonLayout,
  MinistryBrandsSermonList,
  PlusCircle,
  Reverse,
  Settings,
  Spinner,
  StreDown,
  StreUp,
  TextAlignCenter,
  TextAlignLeft,
  TextAlignRight
} from "../../../../icons/src";
import { IconsName } from "../types";

export const Icons = {
  [IconsName.CheckSmall]: CheckSmall,
  [IconsName.MediaImage]: MediaImage,
  [IconsName.MediaVideo]: MediaVideo,
  [IconsName.MediaMap]: MediaMap,
  [IconsName.Heart]: Heart,
  [IconsName.Spinner]: Spinner,
  [IconsName.TextAlignCenter]: TextAlignCenter,
  [IconsName.TextAlignLeft]: TextAlignLeft,
  [IconsName.TextAlignRight]: TextAlignRight,
  [IconsName.PlusCircle]: PlusCircle,
  [IconsName.Reverse]: Reverse,
  [IconsName.Duplicate]: Duplicate,
  [IconsName.CheckCircle]: CheckCircle,
  [IconsName.Settings]: Settings,
  [IconsName.Close]: Close,
  [IconsName.StreUp]: StreUp,
  [IconsName.StreDown]: StreDown,
  [IconsName.Circle2]: Circle2,
  [IconsName.CircleRemove]: CircleRemove,
  [IconsName["t2-sermon-list"]]: MinistryBrandsSermonList,
  [IconsName["t2-sermon-detail"]]: MinistryBrandsSermonDetail,
  [IconsName["t2-sermon-featured"]]: MinistryBrandsSermonFeatured,
  [IconsName["t2-sermon-layout"]]: MinistryBrandsSermonLayout,
  [IconsName["t2-event-list"]]: MinistryBrandsEventList,
  [IconsName["t2-event-detail"]]: MinistryBrandsEventDetail,
  [IconsName["t2-event-featured"]]: MinistryBrandsEventFeatured,
  [IconsName["t2-event-layout"]]: MinistryBrandsEventLayout,
  [IconsName["t2-event-calendar"]]: MinistryBrandsEventCalendar,
  [IconsName["t2-group-list"]]: MinistryBrandsGroupList,
  [IconsName["t2-group-detail"]]: MinistryBrandsGroupDetail,
  [IconsName["t2-group-featured"]]: MinistryBrandsGroupFeatured,
  [IconsName["t2-group-slider"]]: MinistryBrandsGroupSlider,
  [IconsName["t2-group-layout"]]: MinistryBrandsGroupLayout,
  [IconsName["nc-stre-up"]]: StreUp,
  [IconsName["nc-stre-down"]]: StreDown
};
