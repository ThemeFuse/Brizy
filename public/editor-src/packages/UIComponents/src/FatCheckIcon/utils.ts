export const getActiveClassName = (
  className: string,
  isActive?: boolean
): string => `${className}${isActive ? "active" : "inactive"}`;

export const getCheckedClassName = (
  className: string,
  isChecked?: boolean
): string => `${className}${isChecked ? "checked" : "unchecked"}`;
