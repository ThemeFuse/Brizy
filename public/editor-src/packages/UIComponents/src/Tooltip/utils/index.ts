type PositionState = "above" | "below";

const positionState: PositionState = "above";

export const getPosition = (): PositionState => positionState;
