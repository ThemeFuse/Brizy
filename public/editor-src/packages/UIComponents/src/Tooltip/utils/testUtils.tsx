import { ToolbarItemsInstance } from "../types";
import { Placement, Size } from "../types";

export const getToolbar = () =>
  ({
    toolbarRef: { current: document.createElement("div") },
    toolbarCSSPosition: "absolute",
    toolbarItemIndex: 0,
    toolbarItemsLength: 1
  } as unknown as ToolbarItemsInstance);

export const sizes: Size[] = ["auto", "small", "medium", "large", "xlarge"];
export const placements: Placement[] = [
  "auto",
  "auto-start",
  "auto-end",
  "top",
  "top-start",
  "top-end",
  "bottom",
  "bottom-start",
  "bottom-end",
  "right",
  "right-start",
  "right-end",
  "left",
  "left-start",
  "left-end"
];
export const getOffsetPosition = (placement: Omit<Placement, "auto">) => {
  switch (placement) {
    case "right-start":
    case "right-end":
    case "right":
      return "left";
    case "left-start":
    case "left-end":
    case "left":
      return "right";
    case "top-start":
    case "top-end":
    case "top":
      return "bottom";
    case "bottom-start":
    case "bottom-end":
    case "bottom":
      return "top";
  }
};
