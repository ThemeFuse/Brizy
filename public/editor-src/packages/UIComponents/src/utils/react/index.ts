import { SyntheticEvent } from "react";

type WithValue = HTMLElement & { value: string };

export const inputValue = (e: SyntheticEvent<WithValue>): string =>
  (e.target as WithValue).value;
