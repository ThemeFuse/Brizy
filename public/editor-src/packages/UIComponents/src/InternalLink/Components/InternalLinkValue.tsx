import classNames from "classnames";
import React from "react";
import { EditorIcon } from "../../EditorIcon";
import { IconsName } from "../../EditorIcon/types";
import { divStyles } from "../utils";
import { InternalLinkValueProps } from "./types";

export const InternalLinkValue = React.forwardRef<
  HTMLDivElement,
  InternalLinkValueProps
>(({ className, value, onRemove, onClick }, ref) => {
  const _className = classNames(
    "brz-ed-control__internalLink__value",
    className
  );

  return (
    <div className={_className} ref={ref} onClick={onClick} style={divStyles}>
      <span className="brz-span">{value}</span>
      {onRemove && (
        <EditorIcon icon={IconsName.CircleRemove} onClick={onRemove} />
      )}
    </div>
  );
});

InternalLinkValue.displayName = "InternalLinkValue";
