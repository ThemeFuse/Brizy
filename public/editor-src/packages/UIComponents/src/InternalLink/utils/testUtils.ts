import { waitFor } from "@testing-library/react";
import { Choice } from "../types";

export const waitIcon = async (container: HTMLElement) =>
  await waitFor(() =>
    expect(container.querySelector(".brz-icon-svg")).toBeInTheDocument()
  );

export const items: Choice[] = [
  { title: "first", value: 1 },
  { title: "second", value: 2 },
  { title: "third", value: 3 },
  { title: "fourth", value: 4 },
  { title: "fifth", value: 5 },
  { title: "sixth", value: 6 },
  { title: "seventh", value: 7 },
  { title: "eighth", value: 8 },
  { title: "ninth", value: 9 },
  { title: "tenth", value: 10 },
  { title: "eleventh", value: 11 },
  { title: "twelfth", value: 12 },
  { title: "thirteenth", value: 13 },
  { title: "fourteenth", value: 14 },
  { title: "fifteenth", value: 15 }
];
