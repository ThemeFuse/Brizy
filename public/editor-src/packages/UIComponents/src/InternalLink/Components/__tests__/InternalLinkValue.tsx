import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { waitIcon } from "../../utils/testUtils";
import { InternalLinkValue } from "../InternalLinkValue";

describe("Test InternalLinkValue", () => {
  describe("Snapshots and Props", () => {
    test("Without Props", () => {
      const button = render(<InternalLinkValue />);

      expect(button.container.firstChild).toMatchSnapshot();
    });

    test("value should be value", () => {
      const { container } = render(<InternalLinkValue value="value" />);

      const span = container.querySelector<HTMLSpanElement>(".brz-span");

      expect(container.firstChild).toMatchSnapshot();

      expect(span?.innerHTML).toBe("value");
    });

    test("className should be className", () => {
      const { container } = render(<InternalLinkValue className="className" />);

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(".brz-ed-control__internalLink__value")
      ).toHaveClass("className");
    });
  });

  describe("Event Handlers", () => {
    test("Check onClick event handler", () => {
      const onClick = jest.fn();
      const { container } = render(<InternalLinkValue onClick={onClick} />);

      const div = container.querySelector(
        ".brz-ed-control__internalLink__value"
      );

      div && fireEvent.click(div);

      expect(onClick).toHaveBeenCalledTimes(1);
    });

    test("Check onRemove event handler", async () => {
      const onRemove = jest.fn();
      const { container } = render(<InternalLinkValue onRemove={onRemove} />);

      await waitIcon(container);

      const icon = container.querySelector(".brz-icon-svg");

      icon && fireEvent.click(icon);

      expect(onRemove).toHaveBeenCalledTimes(1);
    });
  });
});
