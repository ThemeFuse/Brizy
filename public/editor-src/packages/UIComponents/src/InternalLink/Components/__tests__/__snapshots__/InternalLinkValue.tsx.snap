// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InternalLinkValue Snapshots and Props Without Props 1`] = `
<div
  class="brz-ed-control__internalLink__value"
  style="position: relative;"
>
  <span
    class="brz-span"
  />
</div>
`;

exports[`Test InternalLinkValue Snapshots and Props className should be className 1`] = `
<div
  class="brz-ed-control__internalLink__value className"
  style="position: relative;"
>
  <span
    class="brz-span"
  />
</div>
`;

exports[`Test InternalLinkValue Snapshots and Props value should be value 1`] = `
<div
  class="brz-ed-control__internalLink__value"
  style="position: relative;"
>
  <span
    class="brz-span"
  >
    value
  </span>
</div>
`;
